docs/RFCs/package-toml.md

```diff
@@ -0,0 +1,125 @@
+
+- Version: DRAFT
+- Date: 06/06/2019
+- Authors
+  - Todd Kennedy <todd@selfassembled.org>
+
+## Abstract
+
+This document is meant to describe the package manifest format for Entropic,
+a federated solution to package management. Entropic needs to have its own
+manifest file due to conflicts arising from the namespacing entropic uses.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Copyright
+
+This document is copyright by its individual contributors, and available for
+re-use under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)
+
+## Introduction
+
+Entropic's cannot share a manifest format with the traditional `package.json` format
+associated with the ECMAScript ecosystem for a number of reasons:
+
+1. Entropic is intended to allow for federated package management for more than just the ECMAScript ecosystem
+1. The `package.json` name specifier does not allow for the namespacing that entropic provides
+1. There are recognized deficiencies with using JSON as human-readable metadata format
+
+## Filename
+
+The filename of the manifest **MUST** be `Package.toml` and it **MUST** live in
+the root directory of the package's filesystem. The root of the package is defined
+as the closest parent directory that contains the manifest file, starting with
+the current directory.
+
+On case-insensitive filesystems, it **MAY** be presented as `package.toml` but care
+should be taken to preserve case when possible.
+
+## Format
+
+The format of this file is **REQUIRED** to be in [TOML](https://github.com/toml-lang/toml)
+version 0.5.0 or later as defined by the project.
+
+### Top-Level Data
+
+The manifest **MUST** contain the following top-level keys:
+
+- `"name"` - the canonical name of the package, as a string, including it's namespace. e.g. `"toddself@static-pkg.dev/packge2toml"`
+- `"version"` - a [SEMVER](https://semver.org/) designation conforming to version 2.0.0 of the semantic versioning specification. Due to how TOML parses numbers, this must be represented as a string. e.g. `"1.0.4"`
+
+The manifest **MAY** contain:
+
+- `"entry"` - the filename and package-relative path, as a string, to the main entry point of the package. The root for the relative-path is the directory which contains the `Package.toml` file. e.g `"./src/index.js"`
+- `"type"` - if set to `"module"` this will output a [`package.json` with this included](https://nodejs.org/api/esm.html#esm_code_package_json_code_code_type_code_field), and will signal to the node interpreter to run this code an ES6 Module.
+- `"license"` - a valid [SPDX license](https://spdx.org/licenses/) for your code
+- `"description"` - a short description of your package
+- `"homepage"` - a URL for your project's homepage
+- `"author"` - who wrote this package
```

Comment:
Should a `repository` or `source` optional field be included? That's something I find super valuable with other tools, and always miss it when it's not there.

Reply:
Added repo.  I thought about saying "other fields may appear here, go wild" but then i realized that we might want to not say that in case we need to reserve a field for a future usage that is incompatible with how people are  using it in the wild